### DISCUSSION TITLE
clangd cannot find standard header files

### DISCUSSION THREAD

**Discussion by Proper129** [clangd/clangd#2158] [Q&A] [answered]
clangd will not find any of my standard library header files:
![Screenshot from 2024-09-23 20-16-42](https://github.com/user-attachments/assets/74574882-f0d9-4f85-a908-ea9a1ecb0e8c)

I am on Ubuntu 22.04.4 and using g++ to compile my program.  I have a MakeFile and used the Bear tool to generate the compile_commands.json. This file is in the root of my project. Running make compiles the program with no errors.

During troubleshooting, i used the --check option and get this: [clang.log](https://github.com/user-attachments/files/17106243/clang.log)

It appears to read the compile commands file, but still does not work.

**Comment by HighCommander4** [comment, marked as answer]
Likely duplicate of #1394. Please see workarounds in [this comment](https://github.com/clangd/clangd/issues/1394#issuecomment-1328676884).

**Reply by Proper129** [reply]
This was the problem. I already had g++-12 installed, so I used update-alternatives tool to make g++ command use version 12 over version 11 and everything started working!

Thanks for the help!
